source/table_view.tsx

```diff
@@ -156,24 +155,24 @@ export class TableView extends React.Component<Properties> implements
         return;
       }
     }
-    if(this.columnOrder[0] && this.columnOrder[0].index === index) {
-      this.columnOrder[0] = this.columnOrder[0].reverseSortOrder();
-    } else if( this.columnOrder[0] && -1 <
-        this.columnOrder.findIndex((element) => element.index === index)) {
+    const order = this.table.columnOrder;
+    if(order[0] && order[0].index === index) {
+      order[0] = order[0].reverseSortOrder();
+    } else if(order && -1 <
+        order.findIndex((element) => element.index === index)) {
       const currentIndex =
-        this.columnOrder.findIndex((element) => element.index === index);
-      const order = this.columnOrder.splice(currentIndex);
-      this.columnOrder.unshift(order[0]);
+        order.findIndex((element) => element.index === index);
```

Comment:
This find is being performed twice, factor it out to only perform the search once.